### PR TITLE
Fix ArrayOpsAsyncShift.dropWhile returning head instead of empty when all match

### LINKED ISSUE #123: ArrayOpsAsyncShift.dropWhile returns Array(head) instead of empty when all elements match
`Array(1, 2, 3).dropWhile(_ < 10)` should be empty. Under `async`, the shifted version returns `Array(1)`. The body uses `arr.slice(0, 1)` for the "iterator exhausted" branch where `arr.slice(0, 0)` is intended.

Reproducer (scala-cli):

```scala
//> using scala 3.8.3

@main def repro(): Unit =
  val arr = Array(1, 2, 3)

  // stdlib
  val std = arr.dropWhile(_ < 10)
  println(s"stdlib: ${std.toList}")        // List()
  assert(std.isEmpty)

  // what ArrayOpsAsyncShift.dropWhile does when all match:
  //   if !it.hasNext then monad.pure(arr.slice(0, 1))   // BUG: slice(0,0)
  def buggy(arr: Array[Int], p: Int => Boolean): Array[Int] =
    def skipWhile(it: Iterator[Int], i: Int): Array[Int] =
      if !it.hasNext then arr.slice(0, 1)
      else
        val e = it.next()
        if p(e) then skipWhile(it, i + 1)
        else arr.slice(i, arr.length)
    skipWhile(arr.iterator, 0)

  val out = buggy(arr, _ < 10)
  println(s"shifted: ${out.toList}")       // List(1)
  assert(out.sameElements(Array(1)))

  // partial-match case is fine
  val partial = buggy(Array(1, 5, 2, 3), _ < 3)
  assert(partial.sameElements(Array(5, 2, 3)))
```

Source: https://github.com/dotty-cps-async/dotty-cps-async/blob/636274a3ddb59b0e918b449d04155db7c8081588/shared/src/main/scala/cps/runtime/ArrayOpsAsynsShift.scala#L130-L138

```scala
def dropWhile[F[_]](arr: ArrayOps[A], monad: CpsMonad[F])(p: A => F[Boolean]): F[Array[A]] =
  def skipWhile(it: Iterator[A], i: Int): F[Array[A]] =
    if !it.hasNext then monad.pure(arr.slice(0, 1))     // <- should be (0, 0)
    else
      val e = it.next
      monad.flatMap(p(e)): v =>
        if (v) then skipWhile(it, i + 1) else monad.pure(arr.slice(i, arr.size))
  skipWhile(arr.iterator, 0)
```

**Suggested fix.** Replace `arr.slice(0, 1)` with `arr.slice(0, 0)` (or `Array.empty[A]`). Happy to PR.

### PR DESCRIPTION
**Pull request by rssh**
## Summary
- Fix off-by-one in `ArrayOpsAsyncShift.dropWhile`: the iterator-exhausted branch used `arr.slice(0, 1)` where `slice(0, 0)` was intended, so `dropWhile` returned a one-element prefix instead of an empty array when every element matched the predicate.
- Add regression tests for the all-match and empty-array cases.

Thanks to @haskiindahouse for the [bug report](https://github.com/dotty-cps-async/dotty-cps-async/issues/123) and proposed fix.

Closes #123

## Test plan
- [x] `sbt cpsJVM/testOnly cps.TestCBS1ShiftArrayOps` — all 33 tests pass (including 2 new ones)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: shared/src/test/scala/cps/TestCBS1ShiftArrayOps.scala [hunk 1]
```diff
@@ -81,6 +81,28 @@ class TestCBS1ShiftArrayOps:
...
+  @Test def testDropWhileEmpty(): Unit =
+     val c = async[ComputationBound]{
+               Array.empty[ComputationBound[Int]].dropWhile(x => await(x) < 10)
+             }
+     c.run() match {
+        case Success(v) =>
+            assert(v.isEmpty)
+        case Failure(ex) =>
+            assert(false,"dropWhile on empty array should be successed")
+     }
```
[the remark this answers is not in the document]
Comment (rssh) [reply shared/src/test/scala/cps/TestCBS1ShiftArrayOps.scala:104]: @copilot apply changes based on [this feedback](https://github.com/dotty-cps-async/dotty-cps-async/pull/126#discussion_r3215317059)
